src/extensionWeb.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { GitApiImpl } from './api/api1';
+import Logger from './common/logger';
+import { Resource } from './common/resources';
+import { onceEvent } from './common/utils';
+import * as PersistentState from './common/persistentState';
+import { EXTENSION_ID } from './constants';
+import { } from './github/folderRepositoryManager';
+import { registerLiveShareGitProvider, registerGithubGitProvider } from './gitProviders/api';
+import { PullRequestsTreeDataProvider } from './view/prsTreeDataProvider';
+import { CredentialStore } from './github/credentials';
+import { setTelemetry, aiKey, telemetry, commonDeactivate, init } from './extensionCommon';
+
+export async function activate(context: vscode.ExtensionContext): Promise<GitApiImpl> {
+	// initialize resources
+	Resource.initialize(context);
+	const apiImpl = new GitApiImpl();
+
+	const version = vscode.extensions.getExtension(EXTENSION_ID)!.packageJSON.version;
+	setTelemetry(new TelemetryReporter(EXTENSION_ID, version, aiKey));
+	context.subscriptions.push(telemetry);
+
+	PersistentState.init(context);
+	const credentialStore = new CredentialStore(telemetry);
+	await credentialStore.initialize();
+
+	context.subscriptions.push(registerGithubGitProvider(credentialStore, apiImpl));
```

Comment:
It looks like the only difference between this entry point and `extension.ts` is this registration. I think breaking them up is ok, but one suggestion I have to minimize code duplication is to instead create two versions of `/gitProviders/api` in the `env` folder and have different implementations of `registerBuiltInGitProvider`

Reply:
Done!